tests/_compilertestrunner.livecodescript

```diff
@@ -0,0 +1,681 @@
+      runCompilerTestScript pInfo, tFile
+      put TesterTapCombine(tAnalysis, the result) into tAnalysis
+   end repeat
+   runPrintSummary(tAnalysis)
+
+   return tAnalysis
+end runAllScripts
+
+-- Run the tests found in one specific script file
+private command runCompilerTestScript pInfo, pScriptFile
+   local tTest, tAnalysis
+
+   repeat for each element tTest in listCompilerTestsInFile(pScriptFile)
+      runCompilerTest pInfo, pScriptFile, tTest
+      put TesterTapCombine(tAnalysis, the result) into tAnalysis
+   end repeat
+   return tAnalysis
+end runCompilerTestScript
+
+private command runTestProcessOutput pScriptfile, pCommand, pOutput
+   -- Create test log
+   local tTestLog
+   put "###" && TesterGetPrettyTestName(pScriptFile, "compilertest") && pCommand \
+         & return & return into tTestLog
+   put pOutput & return after tTestLog
+
+   -- Analyse the results and print a summary line
+   local tTapResults
+   put TesterTapAnalyse(tTestLog) into tTapResults
+
+   logSummaryLine tTapResults, (TesterGetPrettyTestName(pScriptFile, "compilertest") & ":" && pCommand)
+
+   return tTapResults
+end runTestProcessOutput
+
+-- Run a specific named test command tCommand in a script file
+-- tScriptFile
+private command runCompilerTest pInfo, pScriptFile, pTest
+   local tCommandLine
+
+   -- First we need to process the specified test in script file
+
+   local tTestInfo
+   processCompilerTest pInfo, pScriptFile, pTest, tTestInfo
+
+   -- Next we emit the code for the test, then attempt to compile it
+   local tTestFile, tTestInterfaceFile
+   put tempName() into tTestFile
+   put tempName() into tTestInterfaceFile
+   put textEncode(tTestInfo["code"], "utf8") into url ("binfile:" & tTestFile)
+   reportCompilerTestDiag format("output test source file to '%s'", tTestFile)
+   reportCompilerTestDiag tTestInfo["code"]
+
+   -- Build the command line
+   local tCompilerCmdLine
+   put format("%s --error-info --interface %s %s", $LC_COMPILE, tTestInterfaceFile, tTestFile) into tCompilerCmdLine
+   reportCompilerTestDiag format("compile command is %s", tCompilerCmdLine)
+
+   -- Try to execute the compilation
+   local tCompilerOutput, tCompilerExitStatus
+   put shell(tCompilerCmdLine) into tCompilerOutput
+   put the result into tCompilerExitStatus
+   reportCompilerTestDiag "compile command output"
+
+   -- Remove the test file
+   delete file tTestFile
+   delete file tTestInterfaceFile
+
+   -- The output from the subprocesses will be native encoded utf-8.
+   put textDecode(tCompilerOutput, "utf8") into tCompilerOutput
+   reportCompilerTestDiag tCompilerOutput
+
+   -- Process the assertions and make sure each one matches up
+   repeat for each line tAssertion in tTestInfo["assertions"]
+      if doesCompilerOutputSatisfyAssertion(tCompilerOutput, tCompilerExitStatus, tAssertion, tTestInfo["positions"]) then
+         put "ok # " after tTestOutput
+      else
+         put "not ok # " after tTestOutput
```

Comment:
I'm not sure this matches the way we generate TAP output elsewhere in the testsuite.

`compilerPrettyPrintAssertion()` appears to return a message describing the test. I think that elsewhere, we use:

```
ok - <description>
not ok - <description>
```

That seems to match my understanding of the TAP specification, too.